Update token guides for Cadence 1.0

joshuahannan opened the pull request:
Updates the ft, nft, and metadata views guides for Cadence 1.0
I'm currently working on updating Chase's example repos, so those should be ready soon.
I'm hoping to get a review on this first though in case there are any changes I need to make to the code

sisyphusSmiling commented on docs/build/guides/fungible-token.md:
```diff
@@ -494,31 +872,34 @@ To learn more about running transactions using CLI, [see the docs](https://devel
 Let's now read the balance of the newly created account (`test-acct`) to check it's zero.
 
-Create this new script file `cadence/scripts/ReadVaultBalance.cdc`:
+Create this new script file `cadence/scripts/get_fooToken_balance.cdc`:
```
I sticking with snake case would be more consistent
```suggestion
Create this new script file `cadence/scripts/get_foo_token_balance.cdc`:
```

joshuahannan replied:
good idea. I'll update all the txs and scripts to have that

sisyphusSmiling commented on docs/build/guides/fungible-token.md:
````diff
@@ -494,31 +872,34 @@ To learn more about running transactions using CLI, [see the docs](https://devel
 
 ```bash
-flow generate script ReadVaultBalance
+flow generate script get_fooToken_balance
````
Follow on from above. Also applies to a couple other places below.

```suggestion
flow generate script get_foo_token_balance
```

sisyphusSmiling commented on docs/build/guides/nft.md:
```diff
@@ -153,7 +130,7 @@ access(all) contract FooBar {
 
     init() {
         self.totalSupply = 0
```
This field is removed above, not sure if it should be here or the intention was to remove it

sisyphusSmiling commented on docs/build/guides/nft.md:
````diff
@@ -183,223 +160,344 @@ access(all) contract FooBar {
 ```
+These interface conformances for [NFT](https://github.com/onflow/flow-nft/blob/master/contracts/NonFungibleToken.cdc#L98)
+and [Collection](https://github.com/onflow/flow-nft/blob/master/contracts/NonFungibleToken.cdc#L190)
+inherit from other interfaces provide important functionality and restrictions
````
```suggestion
inherited from other interfaces provide important functionality and restrictions
```

sisyphusSmiling commented on docs/build/guides/nft.md:
```diff
@@ -183,223 +160,344 @@ access(all) contract FooBar {
+the `withdraw()` function can only be called if the reference 
+is authorized by the owner with `NonFungibleToken.Withdraw`,
+which is [a standard entitlement](https://github.com/onflow/flow-nft/blob/master/contracts/NonFungibleToken.cdc#L58)
+defined by the FungibleToken contract:
```
```suggestion
defined by the NonFungibleToken contract:
```

sisyphusSmiling commented on docs/build/guides/nft.md:
````diff
@@ -473,40 +597,46 @@ flow accounts create
 
 ```bash
-flow generate script GetNFTs
+flow generate script get_fooBar_ids
````
Similar comment as FT guide, snake case would be more consistent IMO

sisyphusSmiling commented on docs/build/guides/nft.md:
```diff
@@ -590,159 +760,28 @@ flow accounts create
 
 ## More
 
 - Explore [an example NFT repository](https://github.com/nvdtf/flow-nft-scaffold/blob/main/cadence/contracts/exampleNFT/ExampleNFT.cdc)
```
We'll want to make sure this repo is updated as wel

sisyphusSmiling reviewed (approved):
Looks great! Left a couple smaller suggestions, mostly just about consistency and linked repos being updated

joshuahannan commented on docs/build/guides/fungible-token.md:
```diff
@@ -42,7 +75,7 @@ If you haven't installed the Flow CLI yet and have [Homebrew](https://brew.sh/)
 ### Initializing a New Project
 
-> 💡 Note: Here is [a link to the completed code](https://github.com/chasefleming/foobar-nft) if you want to skip ahead or reference as you follow along.
+> 💡 Note: Here is [a link to the completed code](https://github.com/chasefleming/FooToken) if you want to skip ahead or reference as you follow along.
```
I'm working on updating that right now. I'll have a PR ready soon

joshuahannan commented on docs/build/guides/fungible-token.md:
```diff
@@ -95,31 +146,45 @@ In this contract file, we want to import our `FungibleToken` contract that we've
-  init() {
-    self.totalSupply = 1000.0
-  }
+    access(all) let VaultStoragePath: StoragePath
+    access(all) let VaultPublicPath: PublicPath
+    access(all) let MinterStoragePath: StoragePath
```
yep, also working on that

joshuahannan commented:
@chasefleming I opened PRs in your repos with the changes from these guides:
* NFT: https://github.com/chasefleming/foobar-nft/pull/1
* FT: https://github.com/chasefleming/FooToken/pull/2

Can you please review this PR and those two PRs so we can get everything merged an updated? Thanks!

chasefleming commented on docs/build/guides/fungible-token.md:
```diff
@@ -17,22 +17,55 @@ Fungible tokens are digital assets that are interchangeable and indistinguishabl
+You can learn more about resources in the Cadence [documentation](https://cadence-lang.org/docs/1.0/language/resources)
+and [tutorials](https://cadence-lang.org/docs/1.0/tutorial/resources).
+
+For Fungible Tokens specifically, tokens are represented by a resource type called a `Vault`:
```
Should "Fungible Tokens" be capitalized here?

chasefleming reviewed (changes requested):
I started noting the sections with line break issues but maybe there is a formatting in your editor to just do it all in one go.

joshuahannan commented:
@chasefleming I addressed your comments and pushed some new changes. As I said in my reply, the line break thing was intentional, so don't worry about that
